ci/authn-gcp/deploy_function.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash -ex
+
+# Script that runs inside a Google SDK container and deploys a function
+# That fetches an id token with audience claim passed in the request query string
+
+WORK_DIR="$HOME"
+GCF_SOURCE_DIR="$WORK_DIR/function"
+GCF_SOURCE_FILE="$GCF_SOURCE_DIR/main.py"
+
+main() {
+  echo "Deploy function: $GCF_FUNC_NAME in project: $GCP_PROJECT"
+  validate_pre_requisites || 1
+  deploy_function || exit 1
+  write_identity_token_to_file || exit 1
+  echo "-> Deploying done"
+}
+
+validate_pre_requisites() {
+  echo 'validate_pre_requisites'
+
+  if [ -z "$GCP_PROJECT" ]; then
+    echo "-- ERROR: function cannot be deployed, GCP project name is undefined."
+    exit 1
+  fi
+
+  if [ -z "$GCF_FUNC_NAME" ]; then
+    echo "-- ERROR: function cannot be deployed, function name is undefined."
+    exit 1
+  fi
+
+  if [ ! -d "$GCF_SOURCE_DIR" ]; then
+    echo "-- ERROR: function cannot be deployed, function directory not found, expected '$GCF_SOURCE_FILE'."
+    exit 1
+  fi
+
+  if [ ! -f "$GCF_SOURCE_FILE" ]; then
+    echo "-- ERROR: function cannot be deployed, function file not found, expected '$GCF_SOURCE_FILE'."
+    exit 1
+  fi
+
+  if [ ! -f "$GCP_OWNER_SERVICE_KEY" ]; then
+    echo "-- ERROR: function cannot be deployed, service account key file not found."
+    exit 1
+  fi
+
+  echo '-> validate_pre_requisites done'
+}
+
+deploy_function() {
+  echo 'deploy_function'
+
+  # Replace the function name with a unique function name in the source code file
+  sed -i "s/func_name/$GCF_FUNC_NAME/" "$GCF_SOURCE_FILE"
```

Comment:
what is motivation behind the name `s/func_name`?

Reply:
s means substitute in sed.